waku/v2/utils/namespacing.nim

```diff
@@ -0,0 +1,54 @@
+## Collection of utilities related to namespaced topics
+## Implemented according to the specified Waku v2 Topic Usage Recommendations
+## More at https://rfc.vac.dev/spec/23/
+
+{.push raises: [Defect]}
+
+import
+  std/strutils,
+  stew/results
+
+type
+  NamespacedTopic* = object
+    application*: string
+    version*: string
+    topicName*: string
+    encoding*: string
+  
+  NamespacingResult*[T] = Result[T, string]
+
+proc fromString*(T: type NamespacedTopic, topic: string): NamespacingResult[NamespacedTopic] =
+  ## Splits a namespaced topic string into its constituent parts.
+  ## The topic string has to be in the format `/<application>/<version>/<topic-name>/<encoding>`
+  
+  let parts = topic.split('/')
+  
+  if parts.len != 5:
+    # Check that we have an expected number of substrings
+    return err("invalid topic format")
+
+  if parts[0] != "":
+    # Ensures that topic starts with a "/"
+    return err("invalid topic format")
+  
+  ok(NamespacedTopic(application: parts[1],
+                     version: parts[2],
+                     topicName: parts[3],
+                     encoding: parts[4]))
+
+proc `$`*(namespacedTopic: NamespacedTopic): string =
+  ## Returns a string representation of a namespaced topic
+  ## in the format `/<application>/<version>/<topic-name>/<encoding>`
+  
+  var topicStr = newString(0)
+
+  topicStr.add("/")
```

Comment:
Just curious why not using `&` in here for string concatenation?

Reply:
Good question! One reason is that using the `add`/append syntax is somewhat more readable for a very structured string, especially where the format may be modified in future.
Another is that `add` modifies the original string (which is the behaviour I want here), whereas `&` creates a new string composed of the two strings being concatenated.